# LOY_OneK1K

This repository contains the complete pipeline used to identify and analyze mosaic loss of chromosome Y (LOY) in immune cells using single-cell RNA-seq data from the OneK1K study.

---

## üî¨ Project Overview

LOY is the most frequent somatic event in aging men and has been associated with increased risk for multiple age-related diseases. In this study, we utilized single-cell transcriptomic data (scRNA-seq) and SNP arrays genotyping to identify LOY at cellular resolution and explore its effects on immune cell phenotypes and gene expression programs.

## required packages:

We used apptainer containers to excute the codes on HPC.

# To process reads, and call LOY from scRNA-seq


2- CellRanger: https://www.10xgenomics.com/support/software/cell-ranger/downloads
3- subset_bam: https://github.com/10XGenomics/subset-bam
4- Velocyto: https://hub.docker.com/r/genomicpariscentre/velocyto
5- python3: https://hub.docker.com/r/jupyter/datascience-notebook/

# To process SNP arrays data, and call LOY from genotyping
1- BCFtools
2- MoChA : https://github.com/freeseek/mocha
3- gtc2vcf: https://github.com/freeseek/gtc2vcf

# To process scRNA-seq data
1- our container with Seurat v4 nd other packages in make_container folder
2- Seurat V5 for DGE with MAST method: https://hub.docker.com/r/satijalab/seurat
3- phate package in python https://pypi.org/project/phate/


---

## üìÅ Repository Structure

| Folder/File | Description |
|-------------|-------------|
| `call_mCA_from_SNParrays/` | Bash scripts for calling mosaic chromosomal alterations (mCA) from SNP array data. |
| `mapping_and_counting/` | Bash scripts and annotations for initial mapping and gene counting steps. |
| `script/` | Miscellaneous annotated R and python scripts used throughout the pipeline. |
| `step_1_count_and_demultiplex_cellranger.sh` | read count processing using Cell Ranger, and demultiplex |
| `step_2_count_velocyto.sh` | Running Velocyto to count spliced/unspliced reads on BAM files generated by CellRanger. |
| `step_3a_call_LOY_cellranger.sh` | Identify LOY cells using Cell Ranger output. |
| `step_3b_call_LOY_velocyto.sh` | Identify LOY cells using Velocyto output. |
| `step_4_merge_cellranger_velocyto_LOY_calls.sh` | Merge LOY calls from Cell Ranger and Velocyto. |
| `step_5_process_barcodes_to_match_rds_file.sh` | Align LOY calls with metadata from Seurat RDS object obtained from CellxGene database. |
| `step_6_prepare_and_collect_all_LOY_calls.sh` | Aggregate LOY calls and prepare for analysis. |
| `README.md` | This file. |

---

## üìä Interactive Results Portal

Explore LOY patterns by donor, cell type, and gene expression using our Shiny app:

üëâ https://lagreen.shinyapps.io/1k1kloy/

---


